src/main/java/com/wojciechbarwinski/demo/epic_board_games_shop/services/OrderProceedAfterPaymentService.java

```diff
@@ -0,0 +1,42 @@
+package com.wojciechbarwinski.demo.epic_board_games_shop.services;
+
+import com.wojciechbarwinski.demo.epic_board_games_shop.clients.LegendaryWarehouseClient;
+import com.wojciechbarwinski.demo.epic_board_games_shop.dtos.OrderDataFromWarehouseDTO;
+import com.wojciechbarwinski.demo.epic_board_games_shop.dtos.OrderDataToWarehouseDTO;
+import com.wojciechbarwinski.demo.epic_board_games_shop.entities.Order;
+import com.wojciechbarwinski.demo.epic_board_games_shop.entities.OrderStatus;
+import com.wojciechbarwinski.demo.epic_board_games_shop.exceptions.OrderNotFoundException;
+import com.wojciechbarwinski.demo.epic_board_games_shop.mappers.MapperFacade;
+import com.wojciechbarwinski.demo.epic_board_games_shop.repositories.OrderRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+class OrderProceedAfterPaymentService {
+
+    private final OrderRepository orderRepository;
+    private final MapperFacade mapper;
+    private final LegendaryWarehouseClient client;
+
+    void proceedOrderAfterPayment(String codeId) {
+        Long id = Long.parseLong(codeId);// method to read codeId from link as order ID
+        Order order = orderRepository.findById(id)
+                .orElseThrow(() -> new OrderNotFoundException(id));
+
+        order.setOrderStatus(OrderStatus.PAID); //tmp useless
+        log.info("Order with id {} was paid", order.getId());
+
+        //TODO ta metoda mi sie nie podoba, robi za dużo rzeczy. Wg mnie powinna po prostu ustawić PAID i tyle. Ale nie wiem jak w takim razie stworzyć to wysłanie danych do LW. Przydałaby się po prostu nowa metoda, ale i tak byłaby tu wywoływana o to brzydkie jakieś takie
+        OrderDataToWarehouseDTO orderDataToWarehouseDTO = mapper.mapOrderToOrderDataToWarehouse(order);
+        log.info("Order with id {} is sending to warehouse...", order.getId());
+
+        OrderDataFromWarehouseDTO orderDataFromWarehouseDTO = client.sendOrderToWarehouse(orderDataToWarehouseDTO);
```

Comment:
W adapterze zróbmy zapis do bazy orderu do wysyłki do magazynu i potem jakiś osobny cron, który podejmie takie zamówienia, wyśle request do magazynu i jeśli się powiedzie to zaktualizuje status orderu